6 заданий по второму занятию.

Eugene62 opened the pull request.

Eugene62 committed:
сделал первое задание

Eugene62 committed:
сделал второе задание

Eugene62 committed:
добавил все задания. тесты выполняются

Eugene62 committed:
поправил табуляцию у if в первом задании

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,22 @@
+"use strict"
+
+function whichType(inputParam) {
+
```
Лишняя строчка.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,22 @@
+
+function whichType(inputParam) {
+
+    let type = typeof(inputParam);
```
Больше похоже на `const`.
`typeof inputParams` -- так принято.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,22 @@
+    let type = typeof(inputParam);
+
+    if ( type === "string" ){
+        return "string";
```
Не стоит плодить строчки, в них легко опечататься, достаточно вернуть `type`, т.к. тут ты уверен, что он 'string'.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,22 @@
+            return "undefined";
+        }
+
+        return "number";
```
Также можно вернуть `type`.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,22 @@
+
+    let type = typeof(inputParam);
+
+    if ( type === "string" ){
```
Правильно так: `if (type === "string") {`, в остальных местах тоже поправь.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,22 @@
+
+    if ( type === "number" ){
+        if ( isNaN( inputParam ) ){
+            return "undefined";
```
Была причина вывести 'undefind' именно строкой?
Последний `return` верен, сделай также.

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,22 @@
+    return undefined;
+}
+
+module.exports = whichType;
```
Про пустые строчки в конце файла не забываем.

djilua1 commented on src/ex2_js-basics-part2/task-02.js:
```diff
@@ -0,0 +1,17 @@
+    else{
+        console.log("Input is not array");
+    }
+}
```
Ну прям никуда стили не годятся...

djilua1 commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,30 @@
+    });
+
+    return [numberOfEven, numberOfOdd, numberOfZeros];
+}
```
Поправь стили.

djilua1 commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,30 @@
+
+    inputArray.forEach(element => {
+
+        if ( element !== null ) {
```
Почему проверка только на `null`?

djilua1 commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,30 @@
+            }
+            else if ( element % 2 === 1 ){
+                ++numberOfOdd;
+            }
```
Ну странно немного, может перепишем?
Смотри, первое условие ок, после него поставим `else`. Далее внутри  блока проверим, что число четное, иначе же оно будет нечетным, другого не дано.

djilua1 commented on src/ex2_js-basics-part2/task-04.js:
```diff
@@ -0,0 +1,18 @@
+    });
+
+    return counter === inputArray.length ? true : false;
+}
```
Поправь стили везде.

djilua1 commented on src/ex2_js-basics-part2/task-04.js:
```diff
@@ -0,0 +1,18 @@
+            
+    });
+
+    return counter === inputArray.length ? true : false;
```
Интересное решение, но не оптимальное, у тебя функция всегда будет пробегать по всему массиву.

djilua1 replied:
Либо переделай в цикл с return, либо используй другой метод массива. :)

djilua1 commented on src/ex2_js-basics-part2/task-05.js:
```diff
@@ -0,0 +1,18 @@
+
+    let maxValue = inputArray[0];
+
+    for (let i = 0; i < inputArray.length; ++i){
```
Мы уже объявили, что нулевой элемент самый большой, может стоит начать с первого?

djilua1 commented on src/ex2_js-basics-part2/task-06.js:
```diff
@@ -0,0 +1,24 @@
+  
+    if ( number <= MAX_VALUE ){
+    
+        const sqrtOfInputNumber = Math.floor(Math.sqrt(number) );
```
А если было введено число 9.3?
Мне кажется это корректные данные, + есть отрицательные числа.
Также 0 и 1 не являются ни простыми, ни составными числами.

djilua1 reviewed (commented):
Пожалуйста, поработай над стилями.

Eugene62 committed:
1) Исправил возвращаемые значения в первом задании
и убрал лишние строки.
2) Поправил стили в задание 2.
3) Поправил стили в задании 3 и убрал лишние условия.
4) Изменил forEach на for c return'ом в 4ом адании.
5) Поправил цикл с первого индекса в 5ом задании.
6) Поправил стили и добавил дополнительные проверки
в 6ом задании.

Eugene62 committed:
поправил замечания по стилям

djilua1 commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,21 @@
+      }
+
+       return type;
+    }
```
Сори, что-то только с твоего PR'a обратил на это внимание.
Если тип число или строка, то проверь, что это не `null` и верни тип. Т.е. таким образом будет не 3 условия, а 2.

djilua1 commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,27 @@
+      if (element === 0) {
+        ++numberOfZeros;
+      }
+      else { 
```
Сползла строчку `} else {`. Ниже та же ошибка.

djilua1 commented on src/ex2_js-basics-part2/task-05.js:
```diff
@@ -0,0 +1,13 @@
+"use strict"
+
+function findMaxInArray(inputArray) {
+  let maxValue = inputArray[0];
```
Отбиваем пустой строкой. :)

djilua1 commented on src/ex2_js-basics-part2/task-05.js:
```diff
@@ -0,0 +1,13 @@
+      maxValue = inputArray[i];
+    }
+  }
+  return maxValue;
```
Тут тоже.

djilua1 commented on src/ex2_js-basics-part2/task-06.js:
```diff
@@ -0,0 +1,20 @@
+
+function isSimple(number) {
+  if (number <= MAX_VALUE && number !== 0 && number !== 1 && number % 1 === 0) {
+    const sqrtOfInputNumber = Math.floor(Math.sqrt(number));
```
Тоже отбиваем. И ниже сдвиг пошел некорректный.

djilua1 commented on src/ex2_js-basics-part2/task-06.js:
```diff
@@ -0,0 +1,20 @@
+const WRONG_INPUT = 'Данные неверны';
+
+function isSimple(number) {
+  if (number <= MAX_VALUE && number !== 0 && number !== 1 && number % 1 === 0) {
```
Проверку на отрицательное число сделай `number <= 1`.

Eugene62 committed:
поправил замечания
